aten/src/ATen/native/Resize.h

```diff
@@ -30,11 +30,13 @@ inline TensorImpl* resize_impl_cpu_(
     return self;
   }
 
-  size_t storage_size = 1;
+  int64_t storage_size = 1;
   if (stride) {
     self->set_sizes_and_strides(size, *stride);
     // NB: storage size can be different from numel.
     for (size_t dim = 0; dim < size.size(); ++dim) {
+      // FIXME: Don't rely on storage_size being negative...
+      // This behavior was carried over from TH
```

Comment:
I want to point out that in chat, you asked a different question:

> is signed integer overflow undefined behavior?

Signed integer overflow is indeed undefined behavior, but **unsigned** integer overflow is well defined, and `size_t` is an unsigned type. Maybe you know this already, in that case, ignore me :)

Reply:
Ah yes I asked the wrong question. I'll change the sizes back to size_t then.

I think the comments still apply  because on a tensor with 0 dimensions each dimension could have stride 0 and we could have positive storage_offset `N`. In this case, this code would resize the storage to size `N`, which is not what we want.